### PR TITLE
added black check in the CI before lint

### LINKED ISSUE #719: Add `black` check to the CI
We could add a check for `black --diff --line-length 110` to check if the code has been formatted correctly.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)

     - Added black format check before lint check in the CI.

* **Please check if the PR fulfills these requirements**

- [x] Closes #719  (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [ ] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [ ] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
+      - name: Format code with black
+        run: |
+          python -m pip install black
+          black --diff --line-length 110 openghg/
```
Comment (gareth-j) [inline .github/workflows/workflow.yaml:33]:
```suggestion
          black --check --line-length 110 openghg/
```

**Review by gareth-j — CHANGES_REQUESTED** [review]
I think we could use `--check`, other than that looks good!

**Comment by SutarPrasad** [comment]
Demo output of the --check for reference.
image

Link to docs for understanding the exit codes : https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#check

**Comment by SutarPrasad** [comment]
> I think we could use `--check`, other than that looks good!

It's done

**Review by gareth-j — APPROVED** [review]
Looks good!

**Pull request merged**